### ISSUE TITLE
GNUbg ID/XGID board positions

### ISSUE THREAD

**Issue by amlwwalker** [foochu/bgweb-api#2]
Can this port interpret XGID board positions - the [GNU docs](https://www.gnu.org/software/gnubg/manual/gnubg.html) suggest GNU backgammon can but wondering if this port includes that?

Would you consider adding it?
Thanks

**Comment by foochu** [comment]
Hi again! 

No, that feature is missing here unfortunately. Without looking at the original code, I'd imagine that is a lot smaller task compared to the cube decision effort. Don't want to promise anything, but who knows if the right inspiration hits ;)

BR,
Rami

**Comment by amlwwalker** [comment]
Hey - thanks for responding to this. I had a quick look and found both:

https://www.gnu.org/software/gnubg/manual/html_node/A-technical-description-of-the-Position-ID.html

and probably more usefully 

https://www.extremegammon.com/extremegammon2.pdf (page 151)
![image](https://user-images.githubusercontent.com/2920180/192825024-54bc0668-600d-40d0-ad6f-7030a62ca831.png)


be cool to add this - as your library can become a basis for upgrading GnuBG (which sadly the original is hard to compile these days, but yours is a good step of the way to an awesome replacement!)

Look forward to hearing if you manage to make any progress - but actually just realised porting the C code is probably easier than trying to write it yourself!

EDIT - I just tried implementing it and its a bit of a nasty brain...mess haha

**Comment by foochu** [comment]
Cool, thanks!! It is useful to understand the theory so can whip up a few unit tests to verify the logic. Super easy to make mistakes while porting. From the outset the C code and Go look the same but there are just enough subtleties to make it a walk on ice 😅

**Comment by amlwwalker** [comment]
haha - personally if you add this feature will help me loads - as I am trying to build up a set of narrated backgammon positions, their equities and "why" thats the best move (in plain english) - it will be super useful to be able to type in the XGID code and get the position in a format I want it (for saving) rather than typing the position in manually
